### PR TITLE
M3/rename

### PR DESCRIPTION
**Pull request by mauro3**
Rename to better names:
- `AbstractStepper` -> `AbstractIntegrator` to avoid confusion between iteration-steps and actual integration methods.  This could also be called `AbstractSolver` but integrator seems nicer.
- `Solver` -> `Problem` because Solver was not a Solver but the problem description.  Note that Chris uses this typename too.
- `DenseStepper` -> `DenseOutput` because it is not an integrator.  TODO: remove subtyping

TODO: if approved, then rename the variable names to reflect the new typenames.

Maybe some convention like this could be good:
- use `t::T, y::Y, dy::Y`
- `prob::Problem, ::P`
- `ivp::IVP, ::O`
- `integ::AbstractIntegrator, ::I`
- `solver::AbstractSolver`, `::S`
- `opts::AbstactOptions`, `::OP`

Proper diff is here: https://github.com/JuliaODE/ODE.jl/pull/9/commits/af8551ad1e7dc838b1b44f9d26eae0332d471704 Not sure why the other commit is showing up.

### CODE REVIEW

**Comment by pwl** [comment]
Hm, I can't say I agree with moving the tableaus to `runge-kutta.jl`.  It clutters the implementation of the actual stepper.  Maybe a compromise: move them to the end of the file?  And if you remove them from `tableaus.jl` there is nothing left there, which isn't a big issue per se but why keep that file anyway?

**Comment by pwl** [comment]
And you seem to define the contents of the tableau in `tableau.jl`, but the actual type is no longer there.

**Comment by mauro3** [comment]
The type def of `TableauRKExplicit` needs to be at the top for it to work.  Should I move the rest down, or maybe best the actual tableaus.

I think the remains of `tableaus.jl` should go into `types.jl`, because that is what is needed to wrap other solvers.  Not sure what you mean with your last comment: https://github.com/JuliaODE/ODE.jl/blob/af8551ad1e7dc838b1b44f9d26eae0332d471704/src/tableaus.jl

**Comment by pwl** [comment]
How about `RKIntegrator<:AbstractIntegrator<:AbstractStepper`? And have `DenseOutput<:AbstractStepper` and use the existing interface?  Also we could rename `Stepper` to `Solver` and `Solver` to `???`?

**Comment by mauro3** [comment]
Yes, that could be good.

**Comment by mauro3** [comment]
Ok, I'll add the suggested sub-typing.  What about the variable names?  And do you prefer `Solver` or `Integrator`?

**Comment by pwl** [comment]
`integ` as a shorthand for `integrator`?

File: examples/ex_iter.jl [hunk 1]
```diff
@@ -11,20 +11,20 @@ opts = Dict(:initstep=>0.1,
             :reltol=>1e-5,
             :abstol=>1e-5)
 # pick your solver
-stepper = [ODE.RKIntegratorAdaptive{:rk45},
```
Comment (pwl) [inline examples/ex_iter.jl]: I would write `solver` here, in principle you could add `DenseOutput` (an `AbstractSolver`) to the list and everything would work.

File: src/ode23s.jl [hunk 2]
```diff
@@ -59,11 +59,11 @@ function show(io::IO, state::RosenbrockState)
 end
 
 
-function init{T}(ode::ExplicitODE{T},
```
Comment (pwl) [inline src/ode23s.jl]: Hm, I don't know about this one, if it were `ivp::IVP` then fine, but `ivp::ExplicitODE` looks strange.  Shouldn't we pretend `ode is an instance of a different type`ExplicitODE`?

Comment (mauro3) [inline src/ode23s.jl]: I paused there too, I'll change it back.

**Comment by mauro3** [comment]
Merge?

**Pull request merged**